### PR TITLE
Update to Bot API 5.0

### PR DESCRIPTION
**Pull request by AndrewLaneX**
# Description

Updated to Bot API 5.0

## Type of change

Please delete options that are not relevant.

- [x] Documentation (typos, code examples or any documentation update)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1570,8 +1570,7 @@ Use this method to edit media of messages sent by the bot or via the bot.
 
 Use this method to edit live location messages sent by the bot or via the bot.
 
-`telegram.editMessageLiveLocation(latitude, longitude, chatId, messageId, inlineMessageId, [markup]) => Promise`
-[Official documentation](https://core.telegram.org/bots/api#editmessagelivelocation)
+`telegram.editMessageLiveLocation(latitude, longitude, chatId, messageId, inlineMessageId, [extra]) => Promise`
```
Comment (wojpawlik) [inline docs/README.md:1573]: Params were reordered
```suggestion
`telegram.editMessageLiveLocation(chatId, messageId, inlineMessageId, latitude, longitude, [extra]) => Promise`
```

File: src/telegram-types.ts [hunk 2]
```diff
@@ -130,6 +134,10 @@ export type ExtraVenue = MakeExtra<
 export type ExtraVideo = MakeExtra<'sendVideo', 'video'>
 export type ExtraVideoNote = MakeExtra<'sendVideoNote', 'video_note'>
 export type ExtraVoice = MakeExtra<'sendVoice', 'voice'>
+export type ExtraCopyMessage = MakeExtra<
+  'copyMessage',
+  'from_chat_id' | 'message_id'
+>
```
Comment (wojpawlik) [inline src/telegram-types.ts:140]: This should go after `ExtraContact` to maintain sorting

File: src/telegram-types.ts [hunk 3]
```diff
@@ -97,6 +97,10 @@ export type ExtraEditMessageText = MakeExtra<
   'editMessageText',
   'message_id' | 'inline_message_id'
 >
+export type ExtraEditMessageLiveLocation = MakeExtra<
+  'editMessageLiveLocation',
+  'message_id' | 'inline_message_id' | 'latitude' | 'longitude'
+>
```
Comment (wojpawlik) [inline src/telegram-types.ts:103]: This should go after `ExtraEditMessageCaption` to maintain sorting

File: src/telegram.ts [hunk 4]
```diff
@@ -89,23 +89,35 @@ class Telegram extends ApiClient {
...
+   * @param dropPendingUpdates Whether to drop pending updates
    */
   setWebhook(
     url: string,
     certificate?: tt.InputFile,
     maxConnections?: number,
-    allowedUpdates?: readonly tt.UpdateType[]
+    allowedUpdates?: readonly tt.UpdateType[],
+    ipAddress?: string,
+    dropPendingUpdates?: boolean
```
Comment (wojpawlik) [inline src/telegram.ts:101]: Optional arguments (especially `boolean`) should be passed as `extra` object.

File: src/telegram.ts [hunk 5]
```diff
@@ -594,9 +617,18 @@ class Telegram extends ApiClient {
...
    * @param chatId Unique identifier for the target group or username of the target supergroup or channel (in the format @username)
    * @param userId Unique identifier of the target user
+   * @param onlyIfBanned Do nothing if the user is not banned
    */
-  unbanChatMember(chatId: number | string, userId: number) {
-    return this.callApi('unbanChatMember', { chat_id: chatId, user_id: userId })
+  unbanChatMember(
+    chatId: number | string,
+    userId: number,
+    onlyIfBanned?: boolean
```
Comment (wojpawlik) [inline src/telegram.ts:625]: Optional arguments (especially `boolean`) should be passed as `extra` object.

File: src/telegram.ts [hunk 6]
```diff
@@ -89,23 +89,35 @@ class Telegram extends ApiClient {
...
     })
   }
 
-  deleteWebhook() {
-    return this.callApi('deleteWebhook', {})
+  /**
+   * Remove webhook integration
+   * @param dropPendingUpdates Whether to drop pending updates
+   */
+  deleteWebhook(dropPendingUpdates?: boolean) {
```
Comment (wojpawlik) [inline src/telegram.ts:117]: Optional arguments (especially `boolean`) should be passed as `extra` object.

**Review by wojpawlik — CHANGES_REQUESTED** [review]
1. Update `typegram` to `^2.3.0`, which added API 5.0 typings.

2. Update readmes to mention API 5.0 support instead of 4.8.

3. Deprecate `sendCopy` in favor of `copyMessage`, perhaps change `sendCopy`'s return type to `Promise<tt.MessageId>` so it can be reimplemented using `copyMessage`.

File: src/telegram.ts (hunk 4 above)
Comment (AndrewLaneX) [reply src/telegram.ts:101]: Should I move all optional aruments to `extra`, or only the new ones?

Comment (wojpawlik) [reply src/telegram.ts:102]: Your call, both approaches have merit

Comment (wojpawlik) [reply src/telegram.ts:102]: Move all.

**Review by BLUE-DEVIL1134 — APPROVED** [review]

File: docs/README.md [hunk 7]
```diff
@@ -2261,13 +2285,14 @@ Specifies an url to receive incoming updates via an outgoing webhook.
 
 Use this method to unban a previously kicked user in a supergroup.
 
-`telegram.unbanChatMember(chatId, userId) => Promise`
+`telegram.unbanChatMember(chatId, userId, [onlyIfBanned]) => Promise`
```
Comment (wojpawlik) [inline docs/README.md:2288]: Last thing:

```suggestion
`telegram.unbanChatMember(chatId, userId, [extra]) => Promise`
```

**Pull request merged**